flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -313,7 +314,12 @@ public void close() {
 			flinkConfig.get(KubernetesConfigOptions.SERVICE_CREATE_TIMEOUT));
 
 		return CompletableFuture.supplyAsync(() -> {
```

Comment:
I think it would be good enough to supply a `SupplierWithException` here via `FunctionUtils.uncheckedSupplier()`.